lib/js/jsfetch.nim

```diff
@@ -0,0 +1,48 @@
+## Fetch
+## =====
+##
+## - Fetch wrapper for JavaScript target: https://developer.mozilla.org/docs/Web/API/Fetch_API
+from asyncjs import PromiseJs
+from httpcore import HttpMethod
+
+when not defined(js) and not defined(nimdoc):
+  {.fatal: "Module jsfetch is designed to be used with the JavaScript backend.".}
+
+type
+  fetchModes* = enum ## JavaScript Fetch API mode options.
+    fmCors, fmNoCors, fmSameOrigin
+  fetchCredentials* = enum ## JavaScript Fetch API Credential options.
+    fcInclude, fcSameOrigin, fcOmit
+  fetchCaches* = enum  ## https://developer.mozilla.org/docs/Web/API/Request/cache
+    fchDefault, fchNoStore, fchReload, fchNoCache, fchForceCache
+  fetchRedirects* = enum ## JavaScript Fetch API Redirects options.
+    frFollow, frError, frManual
+  fetchReferrerPolicies* = enum ## JavaScript Fetch API Referrer Policy options.
+    frpNoReferrer, frpNoReferrerWhenDowngrade, frpOrigin, frpOriginWhenCrossOrigin, frpUnsafeUrl
+
+
+proc fetch*(url: cstring, httpMethod = HttpGET, body = "".cstring,
+  integrity = "".cstring, referrer = "client".cstring, mode = fmNoCors,
+  credentials = fcInclude, cache = fchDefault, redirect = frFollow,
+  referrerPolicy = frpOrigin, keepalive = false): PromiseJs {.importcpp: """(
```

Comment:
This is some serious ``importcpp`` misuse. ;-)

Reply:
Why :grey_question: 
Its just like how other proc on js modules are implemented, and code runs with good performance (I bench'd), code pass js linters (I checked, no warning on chrome/firefox),
it even pass style linters. I try `emit` and `asm` too but they are for different purposes.